internal/webhook/v1alpha1/biosversion_webhook.go

```diff
@@ -0,0 +1,122 @@
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+
+	metalv1alpha1 "github.com/ironcore-dev/metal-operator/api/v1alpha1"
+)
+
+// nolint:unused
+// log is for logging in this package.
+var biosversionlog = logf.Log.WithName("biosversion-resource")
+
+// SetupBIOSVersionWebhookWithManager registers the webhook for BIOSVersion in the manager.
+func SetupBIOSVersionWebhookWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewWebhookManagedBy(mgr).For(&metalv1alpha1.BIOSVersion{}).
+		WithValidator(&BIOSVersionCustomValidator{Client: mgr.GetClient()}).
+		Complete()
+}
+
+// TODO(user): EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+
+// TODO(user): change verbs to "verbs=create;update;delete" if you want to enable deletion validation.
+// NOTE: The 'path' attribute must follow a specific pattern and should not be modified directly here.
+// Modifying the path for an invalid path can cause API server errors; failing to locate the webhook.
+// +kubebuilder:webhook:path=/validate-metal-ironcore-dev-v1alpha1-biosversion,mutating=false,failurePolicy=fail,sideEffects=None,groups=metal.ironcore.dev,resources=biosversions,verbs=create;update,versions=v1alpha1,name=vbiosversion-v1alpha1.kb.io,admissionReviewVersions=v1
+
+// BIOSVersionCustomValidator struct is responsible for validating the BIOSVersion resource
+// when it is created, updated, or deleted.
+//
+// NOTE: The +kubebuilder:object:generate=false marker prevents controller-gen from generating DeepCopy methods,
+// as this struct is used only for temporary operations and does not need to be deeply copied.
+type BIOSVersionCustomValidator struct {
+	Client client.Client
+}
+
+var _ webhook.CustomValidator = &BIOSVersionCustomValidator{}
+
+// ValidateCreate implements webhook.CustomValidator so a webhook will be registered for the type BIOSVersion.
+func (v *BIOSVersionCustomValidator) ValidateCreate(ctx context.Context, obj runtime.Object) (admission.Warnings, error) {
+	biosversion, ok := obj.(*metalv1alpha1.BIOSVersion)
+	if !ok {
+		return nil, fmt.Errorf("expected a BIOSVersion object but got %T", obj)
+	}
+	biosversionlog.Info("Validation for BIOSVersion upon creation", "name", biosversion.GetName())
+
+	biosVersionList := &metalv1alpha1.BIOSVersionList{}
+	if err := v.Client.List(ctx, biosVersionList); err != nil {
+		return nil, fmt.Errorf("failed to list BIOSVersionList: %w", err)
+	}
+
+	for _, bs := range biosVersionList.Items {
+		if biosversion.Spec.ServerRef.Name == bs.Spec.ServerRef.Name {
+			err := fmt.Errorf("server (%v) referred in %v is duplicate of Server (%v) referred in %v",
+				biosversion.Spec.ServerRef,
+				biosversion.Name,
+				bs.Spec.ServerRef,
+				bs.Name,
+			)
+			return nil, apierrors.NewInvalid(
+				schema.GroupKind{Group: biosversion.GroupVersionKind().Group, Kind: biosversion.Kind},
+				biosversion.GetName(), field.ErrorList{field.Duplicate(field.NewPath("spec").Child("ServerRef").Child("Name"), err)})
+		}
+	}
```

Comment:
Would it make sense to put this check into a function and reuse it in `ValidateUpdate()`?